src/core/utils/hooks/useIntersection.tsx

```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+
+export const useIntersection = (
```

Comment:
Maybe add comment explaining what this hook does?

Reply:
Added a comment. Tell me if it is not clear.